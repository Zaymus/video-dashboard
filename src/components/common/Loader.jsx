import styled from "styled-components";

const LoaderContainer = styled.div`
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;

    .spinner {
        animation: spin 1.4s linear infinite;
        max-height: 50px;
        max-width: 50px;
    }

    @keyframes spin {
        100% { transform: rotate(360deg); }
    }

    .path {
        stroke: var(--primary);
        stroke-linecap: round;
        animation: dash 1.5s ease-in-out infinite;
    }

    @keyframes dash {
        0% {
            stroke-dasharray: 1, 150;
            stroke-dashoffset: 0;
        }
        50% {
            stroke-dasharray: 90, 150;
            stroke-dashoffset: -35;
        }
        100% {
            stroke-dasharray: 90, 150;
            stroke-dashoffset: -124;
        }
    }
`;

const Loader = () => {
    return (
        <LoaderContainer role='progressbar'>
            <svg className="spinner" viewBox="0 0 50 50">
                <circle className="path" cx="25" cy="25" r="20" fill="none" strokeWidth="5" />
            </svg>
        </LoaderContainer>
    );
}

export default Loader;