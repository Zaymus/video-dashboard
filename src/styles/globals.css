:root {
  --primary: #1E90FF;
  --secondary: #FF6B6B;
  --background-dark: #121212;
  --surface-dark: #1F1F1F;
  --text-primary-dark: #FFFFFF;
  --text-secondary-dark: #B3B3B3;
  --border-dark: #2C2C2C;
  --success: #2ECC71;
  --error: #E74C3C;
  --warning: #F1C40F;
  --background-light: #F9F9F9;
  --surface-light: #FFFFFF;
  --text-primary-light: #1A1A1A;
  --text-secondary-light: #4F4F4F;
  --border-light: #E0E0E0;

  --videos-gutter-margin: 30px;
  --videos-per-row: 5;
  --videos-usable-width: calc(100% - var(--videos-gutter-margin) * 2);
}

body {
  background-color: var(--background-dark);
  margin: 0;
  color: var(--text-primary-dark);
  font-size: 16pt;
}

#root {
  min-height: 100vh;
}