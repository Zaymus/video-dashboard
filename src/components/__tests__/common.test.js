import { describe, test, expect } from '@jest/globals';
import { render } from '@testing-library/react';

import Loader from '../common/Loader';

describe('Common component tests', () => {
  describe('Loader', () => {
    test('matches snapshot', () => {
      const { container } = render(<Loader />);

      expect(container).toMatchSnapshot();
    })
  });
});