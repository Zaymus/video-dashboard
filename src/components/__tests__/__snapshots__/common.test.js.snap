// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Common component tests Loader matches snapshot 1`] = `
<div>
  <div
    class="sc-bRKDuR fMrgJQ"
    role="progressbar"
  >
    <svg
      class="spinner"
      viewBox="0 0 50 50"
    >
      <circle
        class="path"
        cx="25"
        cy="25"
        fill="none"
        r="20"
        stroke-width="5"
      />
    </svg>
  </div>
</div>
`;
